research/job_referencing/index.md

```diff
@@ -0,0 +1,217 @@
+
+I don't think we need anything more in the database to be able to retrieve the copr build id. We need the data given in the `copr_build` job matching the specified `build_identifier` name in the `tests` job, plus all the data we are already using in the search like the `commit_sha`.
+We need to add the `build_identifier` key in config.
+We need to modify [this code](https://github.com/packit/packit-service/blob/16236087fb8071c83b3802464f3af504f4fb933e/packit_service/worker/handlers/testing_farm.py#L202-L241) in a way that it is able to retrieve the proper `build_id`.
+
+#### `build_identifier` key name or?
+
+Another suggestion was to use `after` as a name for the key. I like it but our implementation, if made as specified above, will not be generic. For this reason I would rather prefer `after_build` or `build_identifier` to mark the fact that in the test job we are waiting for a build job.
+
+### Monorepo: Package dependencies
+
+The following is the [RFE](https://github.com/packit/packit/issues/1903) by LecrisUT
+
+```
+packages:
+A:
+depends:
+- B
+- name: C
+rebuild: true
+B: ...
+C: ...
+```
+
+at first I would simplify it like:
+
+```
+packages:
+A:
+depends:
+- B
+- C
+B: ...
+C: ...
+```
+
+I would always rebuild everything for every package but we can enhance the code and if a package has no changes in the given `commit_sha` and it does not depend on changed packages, we can simply copy lines from the previous database pipeline, filling the database as if the jobs were run.
+
+#### The easiest solution I see following what we are already doing for copr builds and tests
+
+We should [create tasks](https://github.com/packit/packit-service/blob/16236087fb8071c83b3802464f3af504f4fb933e/packit_service/worker/jobs.py#L464-L497) only for those packages that do not depend on other packages.
+When running, as an example, the COPR build for the latest dependent package in the list (C in our example), at the end of the handler we can create the COPR build tasks for package A (if any).
+This can be done for all the kinds of job we directly start at [this lines](https://github.com/packit/packit-service/blob/16236087fb8071c83b3802464f3af504f4fb933e/packit_service/worker/jobs.py#L464-L497).
+
+We can change `bodhi` jobs in a way that they will do nothing, apart from starting new `bodhi` tasks for other packages, unless they belong to a package not required by another package. Ideally only package A, in this example, should do a `bodhi` update if I am not wrong. Or at least we should be able to skip doing something in this job during the chain of dependencies.
+
+I think we don't need changes in the database schema to be able to implement this solution.
+We need to add the `depends` or `depends_on` key to the schema. I slightly prefer the `depends` key, but not strong opinions on that.
+
+### Request new feature: support side tag for multi package update
+
+This is the [RFE](https://github.com/packit/packit/issues/1870).
```

Comment:
I think the key ask is to be able to create an update including builds of multiple packages => we need to react when the last one is available and there will only be a single Bodhi update (per Fedora version).

Sadly, the monorepo way we have does not work here => we need one update for all the packages, not one for each.
A few ideas:
* We can specify the packages to include in a new config option and run the job only the one package (the last to be built, needs to be specified).
* We can run the job for all packages but only the last to be built will actually create the update, others will not do anything. (Seems a bit hackish to me.)

Reply:
I thought this could solve the problem:

>We can change `bodhi` jobs in a way that they will do nothing, apart from 
 starting new `bodhi` tasks for other packages, unless they belong to a 
 package not required by another package. Ideally only package A, in this 
 example, should do a `bodhi` update if I am not wrong. Or at least we should 
 be able to skip doing something in this job during the chain of dependencies.